packages/superset-ui-preset-chart-xy/src/encodeable/types/FieldDef.ts

```diff
@@ -0,0 +1,90 @@
+// eslint-disable no-unused-vars
+import { ValueDef, Value } from 'vega-lite/build/src/fielddef';
+import { Type } from 'vega-lite/build/src/type';
+import { TimeFormatter } from '@superset-ui/time-format';
+import { NumberFormatter } from '@superset-ui/number-format';
+import { WithScale } from './Scale';
+import { WithXAxis, WithYAxis, WithAxis } from './Axis';
+import { WithLegend } from './Legend';
+
+export type Formatter = NumberFormatter | TimeFormatter | ((d: any) => string);
+
+// ValueDef is { value: xxx }
+
+// FieldDef is { field: 'fieldName', type: ... }
```

Comment:
remove since it's defined below? `ValueDef` above seems useful if you are reading code but can't introspect the imported types :)